contracts/core/eaters/Eater.sol

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.19;
+
+import "./IEater.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+abstract contract Eater is Ownable {
+    uint256 internal _rate;
+    address internal wewe;
+
+    function _setRate(uint256 rate) internal {
+        require(rate > 0, "Eater: Rate must be greater than 0");
+
+        if (_rate != rate) {
+            _rate = rate;
+        }
+    }
+
+    function _eat(uint256 amount, address underlying, address from) internal {
+        uint256 weweToTransfer = (amount * _rate) / 100;
+        require(weweToTransfer <= IERC20(wewe).balanceOf(address(this)), "Eater: Insufficient balance to transfer");
+
+        IERC20(underlying).transferFrom(from, address(this), amount);
+        IERC20(wewe).transfer(from, weweToTransfer);
+        emit Eaten(amount, from);
+    }
+
```

Comment:
I think it is better to add a function to withdrawal WEWE from the contract by the owner so that it does not get stuck